File: tests/src/main/java/com/smartamenities/backend/dto/AmenityReplacementResponse.java
package com.smartamenities.backend.dto;

import com.smartamenities.backend.model.AmenityType;
import com.smartamenities.backend.model.OccupancyStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for GET /amenities/replacement response.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AmenityReplacementResponse {
    private String amenityId;
    private String amenityName;
    private AmenityType amenityType;
    private Double distanceMeters;
    private Integer waitTimeMinutes;
    private Integer stallsAvailable;
    private OccupancyStatus occupancyStatus;
}

File: Mobile/app/src/main/java/com/smartamenities/app/model/AmenityReplacementResponse.kt
package com.smartamenities.app.model

data class AmenityReplacementResponse(
    val amenityId: String,
    val amenityName: String,
    val amenityType: String,
    val distanceMeters: Double? = null,
    val waitTimeMinutes: Int? = null,
    val stallsAvailable: Int? = null,
    val occupancyStatus: String? = null
)

